lib/errors.js

```diff
@@ -299,3 +299,15 @@ error.InstanceError = function (message) {
   this.message = message;
 };
 util.inherits(error.InstanceError, error.BaseError);
+
+/**
+ * Thrown when a record was not found, Usually used with kenophobic mode (see message for details)
+ * @extends BaseError
+ * @constructor
+ */
+error.RecordNotFoundError = function (message) {
+  error.BaseError.apply(this, arguments);
+  this.name = 'SequelizeRecordNotFoundError';
+  this.message = message;
+};
```

Comment:
`RecordNotFound` implies single result, maybe a better generic would be `EmptyResultError`